include/cuco/detail/pair.cuh

```diff
@@ -41,10 +46,31 @@ constexpr std::size_t next_pow2(std::size_t v) noexcept {
  * whichever is smaller.
  */
 template <typename First, typename Second>
-constexpr std::size_t pair_alignment() {
+constexpr std::size_t pair_alignment()
+{
   return std::min(std::size_t{16}, next_pow2(sizeof(First) + sizeof(Second)));
 }
-} // namespace detail
+
+template <typename T, typename = void>
+struct is_pair_like_impl : std::false_type {
+};
+
+template <typename T>
+struct is_pair_like_impl<T,
+                         std::void_t<decltype(thrust::get<0>(std::declval<T>())),
+                                     decltype(thrust::get<1>(std::declval<T>()))>>
+  : std::true_type {
+};
```

Comment:
Is this meant to accept `thrust::tuple<int, float, bool>`?

Reply:
No, but every attempt I made to use `std::tuple_size` would break something in Thrust, and `thrust::tuple_size` would break something in `std::`. I suppose I could SFINAE on both of them.